app/controllers/shared/api.php

```diff
@@ -37,41 +37,54 @@
     /*
      * Abuse Check
      */
-    $timeLimit = new TimeLimit($route->getLabel('abuse-key', 'url:{url},ip:{ip}'), $route->getLabel('abuse-limit', 0), $route->getLabel('abuse-time', 3600), $db);
-    $timeLimit->setNamespace('app_'.$project->getId());
-    $timeLimit
-        ->setParam('{userId}', $user->getId())
-        ->setParam('{userAgent}', $request->getUserAgent(''))
-        ->setParam('{ip}', $request->getIP())
-        ->setParam('{url}', $request->getHostname().$route->getPath())
-    ;
-
-    //TODO make sure we get array here
+    $abuseKeyLabel = $route->getLabel('abuse-key', 'url:{url},ip:{ip}');
+    $timeLimitArray = [];
 
-    foreach ($request->getParams() as $key => $value) { // Set request params as potential abuse keys
-        if(!empty($value)) {
-            $timeLimit->setParam('{param-'.$key.'}', (\is_array($value)) ? \json_encode($value) : $value);
-        }
+    if (!is_array($abuseKeyLabel)) {
+        $abuseKeyLabel = [$abuseKeyLabel];
     }
```

Comment:
```suggestion
    $abuseKeyLabel =  (!is_array($abuseKeyLabel)) ? [$abuseKeyLabel] : $abuseKeyLabel;
```

Reply:
This would be an assignment even if it's already an array 🤔